src/main/java/org/wildfly/security/asn1/DERDecoder.java

```diff
@@ -338,6 +349,11 @@ public String decodeObjectIdentifier() throws ASN1Exception {
     }
 
     @Override
+    public BigInteger decodeInteger() throws ASN1Exception {
+        return new BigInteger(drainElementValue());
```

Comment:
I think we need a check here to make sure the next tag is actually INTEGER_TYPE.